3735 new projects should use better default name

dhruvisompura opened the pull request:
### Description
Addresses #3735 

Updating the default name for all projects to use kebab case instead of camel case. Camel case is not a very common format to use for directory names and kebab case should better align with user expectations for directory names.

### Screenshots

**Creating a new project**

https://github.com/user-attachments/assets/30d1be78-f8fd-41b4-ac34-3eb2c91c4222

**Project Name Validation Error**

https://github.com/user-attachments/assets/d2b7ab02-185b-4dd6-b110-51360877eb8c

### QA Notes

- [ ] Verify paths for creating a new project should now use kebab case
  - [ ] Verify the default r project name is: `my-r-project`
  - [ ] Verify the default python project name is: `my-python-project`
  - [ ] Verify the default jupyter notebook name is: `my-jupyter-notebook`

dhruvisompura committed:
Update default project name for R projects (#3735)

Changes the default name used for new R projects
to not utilize lower camel case (atypical for R users to see)

dhruvisompura committed:
Update tests to check for new default R project name

dhruvisompura commented on src/vs/workbench/browser/positronNewProjectWizard/components/steps/projectTypeStep.tsx:
```diff
@@ -18,6 +18,26 @@ import { NewProjectWizardStepProps } from 'vs/workbench/browser/positronNewProje
+ * @param projectType - The type of the project for which to generate a default name.
+ * @returns The default project name as a string.
+ */
+export const getDefaultProjectName = (projectType: NewProjectType) => {
+	const defaultProjectName = localize(
+		'positron.newProjectWizard.projectTypeStep.defaultProjectNamePrefix',
+		"my"
+	);
+	// If the project type is an R Project, the project name is 'my-r-project'.
+	if (projectType === NewProjectType.RProject) {
+		return defaultProjectName + '-' + projectType.toLowerCase().replace(/\s/g, '-');
+	}
+	// The project name for all other project types is 'my' + projectType without spaces, eg. 'myPythonProject'.
+	return defaultProjectName + projectType.replace(/\s/g, '');
+};
```
Open question for the team: What if we changed all of the default project names to be lowercase with dashes, i.e. `my-jupyter-project`, `my-python-project`, etc instead of using `camelCase`.

I don't see `camelCase` formatting used in many other products when it comes to naming directories, files, etc. It would also be a better user experience to have the default project names be consistently formatted. 

Thoughts?

juliasilge replied:
That is a really good point. It was an R user who pointed out how very strange camel case looks to that community,  but it's pretty uncommon for Python to use camel case for projects too. The [most downloaded Python packages](https://pypistats.org/top) (which is the closest analog to "project" I can think of) all use kebab case when needed; there is not a camel case among them.

Let's make all these consistent, `my-jupyter-project` and so forth.

timtmok replied:
It will probably depend on the user what they like. It seems that PEP8 recommends underscore for package and modules. https://peps.python.org/pep-0008/#package-and-module-names

Personally, I think kebab makes sense. I think that's the general naming style for git repos. Python doesn't strike me as a language that has great style convention. PEP8 has a mixture of camel and snake case depending on the context and that seems harder to remember.

dhruvisompura commented:
> Thank you for this PR! We ask that you sign our [Contributor License Agreement](https://posit.co/individual-contributor-agreement/) before we accept your contribution. You can sign the CLA by posting a comment on this PR saying:
> 
> I have read the CLA Document and I hereby sign the CLA
> 
> You can retrigger this bot by commenting **recheck** in this Pull Request. Posted by the **CLA Assistant Lite bot**.

I have read the CLA Document and I hereby sign the CLA

dhruvisompura commented:
recheck

dhruvisompura commented:
I have read the CLA Document and I hereby sign the CLA

dhruvisompura commented:
recheck

timtmok reviewed (dismissed):
Looks good! I'm good with an update here or later for changing Python to kebab as well.

sharon-wang reviewed (commented):
Just one small thing otherwise LGTM!

I agree with Tim that the other project type default names can be updated here or in a separate PR, whichever you prefer 👍

dhruvisompura committed:
Update default project name for all project types

Changes the formatting of the default project names to be kebab case
instead of camel case which is more in line with naming conventions
used by programming languages.

dhruvisompura committed:
Update smoke tests to check for new default project names

dhruvisompura committed:
Update smoke test README

Update the smoke test README to add clarity to the dependency section,
fix typos, and links.

dhruvisompura committed:
Remove export of local function

dhruvisompura commented on src/vs/workbench/browser/positronNewProjectWizard/components/steps/projectTypeStep.tsx:
```diff
@@ -18,6 +18,26 @@ import { NewProjectWizardStepProps } from 'vs/workbench/browser/positronNewProje
+ * @param projectType - The type of the project for which to generate a default name.
+ * @returns The default project name as a string.
+ */
+export const getDefaultProjectName = (projectType: NewProjectType) => {
```
Thanks for catching that!

sharon-wang reviewed (approved):
LGTM! Thanks for the smoke test README fixes too 🙌 

Could you please update the PR title/description to reflect that all project type default names have been updated, so our QA team will know to verify all three?
